include/seqan3/utility/bloom_filter/bloom_filter.hpp

```diff
@@ -0,0 +1,385 @@
+    size_t count(value_range_t && values) const noexcept
+    {
+        static_assert(std::ranges::input_range<value_range_t>, "The values must model input_range.");
+        static_assert(std::unsigned_integral<std::ranges::range_value_t<value_range_t>>,
+                      "An individual value must be an unsigned integral.");
+
+        size_t result = 0;
+
+        for (auto && value : values)
+            result += contains(value);
+
+        return result;
+    }
+    //!\}
+
+    /*!\name Capacity
+     * \{
+     */
+    /*!\brief Returns the number of hash functions used in the Bloom Filter.
+     * \returns The number of hash functions.
+     */
+    size_t hash_function_count() const noexcept
+    {
+        return hash_funs;
+    }
+
+    /*!\brief Returns the size that the Bloom Filter manages.
+     * \returns The size in bits.
+     */
+    size_t size() const noexcept
+    {
+        return size_in_bits;
+    }
+    //!\}
+
+    /*!\name Comparison operators
+     * \{
+     */
+    /*!\brief Test for equality.
+     * \param[in] lhs A `seqan3::bloom_filter`.
+     * \param[in] rhs `seqan3::bloom_filter` to compare to.
+     * \returns `true` if equal, `false` otherwise.
+     */
+    friend bool operator==(bloom_filter const & lhs, bloom_filter const & rhs) noexcept
+    {
+        return std::tie(lhs.size_in_bits, lhs.hash_shift, lhs.hash_funs, lhs.data) ==
+               std::tie(rhs.size_in_bits, rhs.hash_shift, rhs.hash_funs, rhs.data);
+    }
+
+    /*!\brief Test for inequality.
+     * \param[in] lhs A `seqan3::bloom_filter`.
+     * \param[in] rhs `seqan3::bloom_filter` to compare to.
+     * \returns `true` if unequal, `false` otherwise.
+     */
+    friend bool operator!=(bloom_filter const & lhs, bloom_filter const & rhs) noexcept
+    {
+        return !(lhs == rhs);
+    }
+    //!\}
+
+    /*!\name Access
+     * \{
+     */
+    /*!\brief Provides direct, unsafe access to the underlying data structure.
+     * \returns A reference to an SDSL bitvector.
+     *
+     * \details
+     *
+     * \noapi{The exact representation of the data is implementation defined.}
+     */
+    constexpr data_type & raw_data() noexcept
+    {
+        return data;
+    }
+
+    //!\copydoc raw_data()
+    constexpr data_type const & raw_data() const noexcept
+    {
+        return data;
+    }
+    //!\}
+
+    /*!\cond DEV
+     * \brief Serialisation support function.
+     * \tparam archive_t Type of `archive`; must satisfy seqan3::cereal_archive.
+     * \param[in] archive The archive being serialised from/to.
+     *
+     * \attention These functions are never called directly, see \ref serialisation for more details.
+     */
+    template <cereal_archive archive_t>
+    void CEREAL_SERIALIZE_FUNCTION_NAME(archive_t & archive)
+    {
+        archive(size_in_bits);
+        archive(hash_shift);
+        archive(hash_funs);
+        archive(data);
+    }
+    //!\endcond
+};
+
+} // namespace seqan3
```

Comment:
```suggestion
};

//!\}

} // namespace seqan3

```

The `//!\}` closes the `addtogroup submodule_bloom_filter` from the top, and the newline at the end was missing